app/Http/Controllers/ReportCrudController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Transaction;
+use App\Services\UserService;
+use Backpack\CRUD\app\Http\Controllers\CrudController;
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanel;
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;
+
+/**
+ * Class SubjectCategoryCrudController
+ *
+ * @property-read CrudPanel $crud
+ */
+class ReportCrudController extends CrudController
+{
+    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
+    use \Backpack\Pro\Http\Controllers\Operations\InlineCreateOperation;
+    
+    public function setup(): void
+    {
+        CRUD::setModel(\App\Models\Transaction::class); // Transaction
+        CRUD::setRoute(config('backpack.base.route_prefix').'/report');
+        CRUD::setEntityNameStrings('transaction', 'transactions');
+
+        if (!UserService::hasAccessTo('bookkeeping')) {
+            $this->crud->denyAllAccess();
+        }
+    }
+
+    protected function setupListOperation(): void
+    {
+        CRUD::column('name');
+        CRUD::column('description');
+    }
+
+    protected function setupCreateOperation(): void
+    {
+        CRUD::setValidation(\App\Http\Requests\ReportRequest::class);
+
+        CRUD::field('name');
+        CRUD::field('description');
+    }
+
+    protected function setupUpdateOperation(): void
+    {
+        $this->setupCreateOperation();
+    }
+    
+    public function showReport()
+    {
+        $incomeCategories = [1, 3, 4];
```

Comment:
This should be built dynamically. It is not a good idea to hardcode this list. What if we add new categories in the future? Then we will need to remember to come here and add them manually. Chances are we will never remember it.

Use the `TransactionCategory` Model to get the right ids.